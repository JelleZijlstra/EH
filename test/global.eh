#!/usr/bin/ehi
# More tests for the global keyword
class Foo
	public bar
	public baz = func: n
		put(n.toString() + ' ')
		echo myvar
	end
end
myvar = 42
Foo.baz 'test'
myvar = 'foo'
Foo.baz 'bah'
newvar = Foo.baz
newvar 19
put(19.toString())
echo(19.toString())
func unknownglobal:
	printvar nosuchvar
end
unknownglobal ()
