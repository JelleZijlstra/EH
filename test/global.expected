test 42
bah foo
19 foo
1919
Uncaught exception of type NameError: Unknown member nosuchvar in object of type Function: 0x10021a588
